config/environments/production.php

```diff
@@ -6,11 +6,22 @@
 /** Disable all file modifications including updates and update notifications */
 define('DISALLOW_FILE_MODS', true);
 
-define('S3_UPLOADS_BUCKET', env('S3_UPLOADS_BUCKET'));
-define('S3_UPLOADS_KEY', env('S3_UPLOADS_KEY'));
-define('S3_UPLOADS_SECRET', env('S3_UPLOADS_SECRET'));
-define('S3_UPLOADS_REGION', env('S3_UPLOADS_REGION'));
-define('S3_UPLOADS_ENDPOINT', env('S3_UPLOADS_ENDPOINT'));
+define('S3_UPLOADS_BUCKET', getenv('S3_UPLOADS_BUCKET'));
+define('S3_UPLOADS_KEY', getenv('S3_UPLOADS_KEY'));
+define('S3_UPLOADS_SECRET', getenv('S3_UPLOADS_SECRET'));
+define('S3_UPLOADS_REGION', getenv('S3_UPLOADS_REGION'));
+define('S3_UPLOADS_ENDPOINT', getenv('S3_UPLOADS_ENDPOINT'));
+if (getenv('S3_UPLOADS_ENDPOINT') && getenv('S3_UPLOADS_BUCKET') && !getenv('S3_UPLOADS_BUCKET_URL')) {
+    $s3_uploads_endpoint = getenv('S3_UPLOADS_ENDPOINT');
+    $s3_uploads_bucket = getenv('S3_UPLOADS_BUCKET');
+    $parts = parse_url($s3_uploads_endpoint);
+    $bucket_url = $parts['scheme'] . "://" . $s3_uploads_bucket . "." . $parts['host'];
+
+    // Define the base bucket URL (without trailing slash)
+    define('S3_UPLOADS_BUCKET_URL', $bucket_url);
+} else {
```

Comment:
The `S3_UPLOADS_BUCKET` will probably be used by the plugin. S3_UPLOADS_BUCKET_URL is already defined in the plugin to `https://BUCKET_NAME.s3.amazonaws.com` => https://github.com/humanmade/S3-Uploads/blob/d3681c3b35cd4da439efdf62ed70898dbab7cb3c/inc/class-plugin.php#L227

There is no problem in using all the variables together.

Reply:
The env var `S3_UPLOADS_BUCKET_URL` is only there if the default system does not work. It will not be required and used by default. It's an optional env var.

Reply:
I'm sorry, I want to accept this PR but I don't understand what you say. I mean, what you say is that if the environment variable `S3_UPLOADS_BUCKET` is defined, the environment variable `S3_UPLOADS_BUCKET_URL` is ignored. Then there is something that is definitely confusing as I said. Did I miss something?

We can talk about it on Discord if you prefer

Reply:
The variable `S3_UPLOADS_BUCKET_URL` is ignored by default. And the default value of this setting is defined in the plugin. If the customer has a specific use, for example he would like to use `https://bucket.s3.fr-par.scw.cloud/directory/subdirectory` instead of `https://bucket.s3.fr-par.scw.cloud` (the default one used when `S3_UPLOADS_ENDPOINT` is defined to `s3.fr-par.scw.cloud` and `S3_UPLOADS_BUCKET` to `bucket`).

This allows all types of bucket configuration to be accepted without changing the entire functionality.
If I haven't made it sufficiently clear, we can discuss this directly in the office.